tqecjs/src/workspace/plaquettes/PlaquetteClass.js

```diff
@@ -0,0 +1,370 @@
+			if (Math.abs(deltaX) > Math.abs(deltaY)) {
+				// Dragging horizontally
+				if (deltaX > 0) {
+					// Moving to the right
+					shiftQ = this.mostRightQubit();
+					// Find the neighboring qubit that is closest to the right
+					const newrq = shiftQ.neighbors.find(
+						(qubit) =>
+							qubit.globalX > shiftQ.globalX && qubit.globalY === shiftQ.globalY
+					);
+
+					diff = newrq.globalX - shiftQ.globalX;
+				} else {
+					// Generate the qubits that are closest to the left
+					shiftQ = this.mostLeftQubit();
+					// Find the neighboring qubit that is closest to the left
+					const newlq = shiftQ.neighbors.find(
+						(qubit) =>
+							qubit.globalX < shiftQ.globalX && qubit.globalY === shiftQ.globalY
+					);
+					diff = newlq.globalX - shiftQ.globalX;
+				}
+				// Shift the qubits by the difference
+				for (const qubit of this.qubits) {
+					const q = qubit.neighbors.find(
+						(q) =>
+							q.globalX === qubit.globalX + diff && q.globalY === qubit.globalY
+					);
+					newQubits.push(q);
+				}
+			} else {
+				if (deltaY > 0) {
+					// Generate the qubits that are closest to the top
+					shiftQ = this.mostTopQubit();
+					console.log(shiftQ, 'shiftQ tq');
+					console.log(shiftQ.neighbors);
+					// Find the neighboring qubit that is closest to the top
+					const newtq = shiftQ.neighbors.find(
+						(qubit) =>
+							qubit.globalX === shiftQ.globalX && qubit.globalY > shiftQ.globalY
+					);
+
+					diff = newtq.globalY - shiftQ.globalY;
+				} else {
+					// Generate the qubits that are closest to the bottom
+					shiftQ = this.mostBottomQubit();
+					// Find the neighboring qubit that is closest to the bottom
+					const newbq = shiftQ.neighbors.find(
+						(qubit) =>
+							qubit.globalX === shiftQ.globalX && qubit.globalY < shiftQ.globalY
+					);
+					diff = newbq.globalY - shiftQ.globalY;
+				}
+				// Shift the qubits by the difference
+				for (const qubit of this.qubits) {
+					const q = qubit.neighbors.find(
+						(q) =>
+							q.globalX === qubit.globalX && q.globalY === qubit.globalY + diff
+					);
+					newQubits.push(q);
+				}
+			}
+
+			// Create a new plaquette
+			let color = 'yellow';
+			if (this.color !== 'purple') {
+				color = 'purple';
+			}
+			console.log(color);
+			const newPlaquette = new Plaquette(
+				newQubits,
+				this.workspace,
+				this.gridSize,
+				color
+			);
+			// Change the color of the plaquette
+
+			// Add the plaquette to the tile
+			this.addChild(newPlaquette);
+		}
+	};
+
+	changePlaquetteColor(newColor) {
+		this.color = newColor;
+		// Update the color of the convex hull shape.
+		this.clear();
+		this.createPlaquette();
+	}
+
+	onDragEnd() {
+		this.isDragging = false;
+		this.initialPosition = null;
+	}
+
+	toggleCtrlButtons = () => {
+		let lastClickTime = 0;
+		this.on('click', (e) => {
+			const currentTime = Date.now();
+			if (currentTime - lastClickTime < 300) {
+				// Set the alpha of the plaquette to 0.8 to indicate
```

Comment:
Indicate that it's selected?

Reply:
yea i still need to implement an alpha change and definitely will. just am still thinking about where to put new plaquettes in with respect to the pixi stage.